src/main/java/org/jpeek/metrics/cohesion/LCOM5.java

```diff
@@ -0,0 +1,184 @@
+import java.util.Set;
+import javassist.CannotCompileException;
+import javassist.CtClass;
+import org.jpeek.Base;
+import org.jpeek.Metric;
+import org.jpeek.metrics.Colors;
+import org.jpeek.metrics.JavassistClasses;
+import org.objectweb.asm.ClassReader;
+import org.objectweb.asm.ClassVisitor;
+import org.objectweb.asm.FieldVisitor;
+import org.objectweb.asm.MethodVisitor;
+import org.objectweb.asm.Opcodes;
+import org.objectweb.asm.Type;
+import org.xembly.Directive;
+
+/**
+ * Lack of Cohesion Of Methods [Henderson & Sellers] (LCOM5).
+ *
+ * <p>This metric shows a ratio between number of methods which
+ * use fields of a class and number of all methods in a class.
+ * Better when all methods use all fields or when class has
+ * number of methods greater than number of fields and all of
+ * these methods use all fields. These one classes have higher
+ * cohesion against classes where each method use only one
+ * field of a class or doesn't use any field.</p>
+ *
+ * <p>The metric value ranges between 0 and any possible number.
+ * The closer to 0 value represents higher cohesion.</p>
+ *
+ * @author Sergey Karazhenets (sergeykarazhenets@gmail.com)
+ * @version $Id$
+ * @see <a href="http://waset.org/publications/5239/a-design-based-cohesion-metric-for-object-oriented-classes">
+ *   A Design-Based Cohesion Metric for Object-Oriented Classes</a>
+ * @see <a href="http://www.math.md/files/csjm/v25-n1/v25-n1-(pp44-74).pdf">
+ *   Class Cohesion Metrics for Software Engineering: A Critical Review</a>
+ * @see <a href="https://www.google.by/url?sa=t&rct=j&q=&esrc=s&source=web&cd=3&ved=0ahUKEwjh9PK4nZ7XAhVlEpoKHRMTAS0QFgg2MAI&url=http%3A%2F%2Fwww.ece.rutgers.edu%2F~marsic%2Fbooks%2FSE%2Finstructor%2Fslides%2Flec-16%2520Metrics-Cohesion.ppt&usg=AOvVaw3967dvApLWjTkj6-MBEnkS">
+ *   Lecture 16: Class Cohesion Metrics</a>
+ * @since 0.13
+ * @checkstyle AbbreviationAsWordInNameCheck (5 lines)
+ * @checkstyle ClassDataAbstractionCouplingCheck (500 lines)
+ */
+public final class LCOM5 implements Metric {
+
+    /**
+     * The base.
+     */
+    private final Base base;
+
+    /**
+     * Ctor.
+     * @param bse The base
+     */
+    public LCOM5(final Base bse) {
+        this.base = bse;
+    }
+
+    @Override
+    public Iterable<Directive> xembly() throws IOException {
+        return new JavassistClasses(
+            this.base, LCOM5::cohesion,
+            // @checkstyle MagicNumberCheck (1 line)
+            new Colors(0.67d, 0.42d)
+        ).xembly();
+    }
+
+    /**
+     * Calculates LCOM5 metric for a single Java class.
+     *
+     * @param ctc The .class file.
+     * @return Metrics.
+     * @checkstyle CyclomaticComplexityCheck (10 lines)
+     * @checkstyle ReturnCountCheck (50 lines)
+     * @checkstyle AnonInnerLengthCheck (50 lines)
+     * @checkstyle ParameterNumberCheck (50 lines)
+     * @checkstyle MethodBodyCommentsCheck (50 lines)
+     */
+    @SuppressWarnings({"PMD.UseVarargs", "PMD.UselessParentheses",
```

Comment:
@sergey-karazhenets why `"PMD.UselessParentheses"`? Can't you simply get rid of parentheses?